jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocInfo.java

```diff
@@ -214,6 +214,12 @@ public Cluster findClusterInPool(String filter) throws SQLException {
                     clusterControllerClient
                             .listClusters(params.projectId(), params.region(), filter)
                             .iterateAll()) {
+
+                // Only connect to cluster with status state `RUNNING` or `UPDATING`
```

Comment:
If you can use a Java 8 filter, that might be cleanerr.

As discussed offline, consider constructing `allClusters` first (as the result of that filtering), and then constructing clusterLoads after that.